src/ngraph/pass/constant_folding_shape_of.cpp

```diff
@@ -35,7 +36,13 @@ void pass::ConstantFolding::construct_constant_shape_of()
 
         auto arg_match = pattern_value_map[arg_label];
 
-        if (arg_match.get_partial_shape().is_static())
+        auto partial_shape = arg_match.get_partial_shape();
+        auto original_shape_of_node = as_type_ptr<op::ShapeOf>(m.get_match_root());
+        if (!original_shape_of_node->is_foldable())
+        {
+            return true;
```

Comment:
I'm not sure but I think we should return false if constant folding didnt occur.